recipes/_compute_slurm_config.rb

```diff
@@ -22,6 +22,14 @@
   action %i[mount enable]
 end
 
+# Check to see if there is GPU on the instance, only execute run_nvidiasmi if there is GPU
+has_gpu = `lspci | grep -i -o 'NVIDIA'`
```

Comment:
what is this for?

Reply:
we need to run `nvidia-smi` on the compute nodes at least once so that the GPU devices are mounted in `/dev/nvidia0` etc.

Reply:
with `if !has_gpu.strip.empty?` aren't we just checking that the string is not empty and not the result of the command?

Reply:
Yes, we are only checking if the string is empty. If the instance does not have GPU the grep on `lspci` will fail and return an empty string, if instance has at least 1 GPU the grep will return something. So if the grep is not empty(instance has GPU) we run `nvidia-smi` which will automatically mount the GPU devices so they are available for use.

Reply:
what if instance has GPU but Nvidia drivers are not installed, e.g. creation of a custom AMI without drivers installation? Can the execution of `nvidia-smi` skipped in this case?